```
======================================================================

FAIL: test_n_chunks (pyemma.coordinates.tests.test_coordinates_iterator.DCD)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/Users/travis/miniconda/envs/_test/lib/python2.7/site-packages/pyEMMA-2.2.2+65.ge9903fe-py2.7-macosx-10.6-x86_64.egg/pyemma/coordinates/tests/test_coordinates_iterator.py", line 83, in test_n_chunks

    self.assertEqual(chunks, it._n_chunks, "stride={s}, lag={t}".format(s=stride, t=lag))
E               AssertionError: 3 != 6 : stride=2, lag=0

```

However this only fails for DCD... The expectation, that we would get one chunk of size 30 and one of size 20 per trajectory does not hold. But how the heck can it hold for the other formats?! It uses exactly the same infrastructure.

--- a/doc/source/CHANGELOG.rst
+++ b/doc/source/CHANGELOG.rst
@@ -6,6 +6,9 @@ Changelog
 
 **New features**:
 
+- coordinates: readers now have the ability to skip the initial n frames. The skip
+  parameter of iterators will be added up to the prior one.
+
 
 **Fixes**:
 

--- a/pyemma/coordinates/data/feature_reader.py
+++ b/pyemma/coordinates/data/feature_reader.py
@@ -350,7 +350,7 @@ class FeatureReaderIterator(DataSourceIterator):
             """ in case the underlying mdtraj iterator raises StopIteration (eg. seek failed),
                 we have to return an empty iterable, so that LaggedIterator will continue to process.
             """
-            if si.args and "too short" in si.args[0] and self._itraj < self._data_source.ntraj - 1:
+            if si.args and ("eof" in si.args[0] or "too short" in si.args[0]) and self._itraj < self._data_source.ntraj - 1:
                 self._next_file()
                 return ()
             else:

--- a/pyemma/coordinates/util/patches.py
+++ b/pyemma/coordinates/util/patches.py
@@ -197,8 +197,12 @@ class iterload(object):
             return next(self._ra_it)
         else:
             if self._extension not in _TOPOLOGY_EXTS:
-                traj = self._f.read_as_traj(self._topology, n_frames=self._chunksize * self._stride,
-                                            stride=self._stride, atom_indices=self._atom_indices, **self._kwargs)
+                if self._extension in ('.dcd'):
+                    traj = self._f.read_as_traj(self._topology, n_frames=self._chunksize,
+                                                stride=self._stride, atom_indices=self._atom_indices, **self._kwargs)
+                else:
+                    traj = self._f.read_as_traj(self._topology, n_frames=self._chunksize * self._stride,
+                                                stride=self._stride, atom_indices=self._atom_indices, **self._kwargs)
             else:
                 traj = self._f.read_as_traj(n_frames=self._chunksize * self._stride,
                                             stride=self._stride, atom_indices=self._atom_indices, **self._kwargs)

